### PR TITLE
Add logs query options

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.Query/api/Azure.Monitor.Query.netstandard2.0.cs [hunk 1]
```diff
@@ -59,6 +65,7 @@ public partial class LogsQueryResult
     {
         internal LogsQueryResult() { }
         public Azure.Monitor.Query.Models.LogsQueryResultTable PrimaryTable { get { throw null; } }
+        public System.Text.Json.JsonElement Statistics { get { throw null; } }
```
Comment (pakrym) [inline sdk/monitor/Azure.Monitor.Query/api/Azure.Monitor.Query.netstandard2.0.cs:68]: Not a final API shape. Swagger would be updated but I wanted to have something to tests.

File: sdk/monitor/Azure.Monitor.Query/src/LogsClient.cs [hunk 2]
```diff
@@ -87,15 +89,80 @@ public virtual LogsBatchQuery CreateBatchQuery()
...
+        internal static QueryBody CreateQueryBody(string query, TimeSpan? timeSpan, LogsQueryOptions options, out string prefer)
         {
             var queryBody = new QueryBody(query);
             if (timeSpan != null)
             {
                 queryBody.Timespan = TypeFormatters.ToString(timeSpan.Value, "P");
             }
 
+            StringBuilder preferBuilder = null;
+            if (options?.Timeout is TimeSpan timeout)
```
Comment (christothes) [inline sdk/monitor/Azure.Monitor.Query/src/LogsClient.cs:101]: What else can `options.Timeout` be?

File: sdk/monitor/Azure.Monitor.Query/src/LogsClient.cs [hunk 3]
```diff
@@ -87,15 +89,80 @@ public virtual LogsBatchQuery CreateBatchQuery()
...
-        internal static QueryBody CreateQueryBody(string query, TimeSpan? timeSpan)
+        internal static QueryBody CreateQueryBody(string query, TimeSpan? timeSpan, LogsQueryOptions options, out string prefer)
         {
             var queryBody = new QueryBody(query);
             if (timeSpan != null)
             {
                 queryBody.Timespan = TypeFormatters.ToString(timeSpan.Value, "P");
             }
 
+            StringBuilder preferBuilder = null;
```
Comment (christothes) [inline sdk/monitor/Azure.Monitor.Query/src/LogsClient.cs:100]: For such a small string, allocating a `StringBuilder` is probably slower than interpolation or string concat.

File: sdk/monitor/Azure.Monitor.Query/src/LogsClient.cs [hunk 4]
```diff
@@ -87,15 +89,80 @@ public virtual LogsBatchQuery CreateBatchQuery()
...
+            {
+                throw new ArgumentNullException(nameof(workspaceId));
+            }
+
+            QueryBody queryBody = CreateQueryBody(query, timeSpan, options, out string prefer);
+            using var message = _queryClient.CreateExecuteRequest(workspaceId, queryBody, prefer);
+
+            if (options?.Timeout != null)
+            {
+                message.SetProperty("NetworkTimeoutOverride", options.Timeout);
```
Comment (christothes) [inline sdk/monitor/Azure.Monitor.Query/src/LogsClient.cs:131]: Should we make "NetworkTimeoutOverride" a known const somewhere?

File: sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientClientLiveTests.cs [hunk 5]
```diff
@@ -428,6 +429,68 @@ public async Task ThrowsExceptionWhenBatchQueryNotFound()
...
+                Assert.AreEqual(JsonValueKind.Undefined, result.Statistics.ValueKind);
+            }
+        }
+
+        [RecordedTest]
+        public async Task CanSetServiceTimeout()
+        {
+            var client = CreateClient();
+
+            // TODO: How to test this?
```
Comment (christothes) [inline sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientClientLiveTests.cs:484]: Would we want to test this with a mock transport in a non-recorded test just to validate that we make the request with the timeout value set?

File: sdk/monitor/Azure.Monitor.Query/src/LogsClient.cs (hunk 2 above)
Comment (pakrym) [reply sdk/monitor/Azure.Monitor.Query/src/LogsClient.cs:101]: `null`

Comment (christothes) [reply sdk/monitor/Azure.Monitor.Query/src/LogsClient.cs:101]: Why not `HasValue` rather than type checking?

Comment (pakrym) [reply sdk/monitor/Azure.Monitor.Query/src/LogsClient.cs:101]: Shorter usage `timeout` vs `options?.Timeout.Value`

but also being cool.

Comment (pakrym) [reply sdk/monitor/Azure.Monitor.Query/src/LogsClient.cs:101]: I kind of like this pattern of narrowing down a value and giving it an alias.

File: sdk/monitor/Azure.Monitor.Query/src/LogsClient.cs (hunk 4 above)
Comment (pakrym) [reply sdk/monitor/Azure.Monitor.Query/src/LogsClient.cs:127]: I'll turn it into an actual property.

**Review by christothes — APPROVED** [review]

**Pull request merged**